cache/topic.go

```diff
@@ -0,0 +1,110 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	dpcache "github.com/ONSdigital/dp-cache"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// TopicCache is a wrapper to dpcache.Cache which has additional fields and methods specifically for caching topics
+type TopicCache struct {
+	*dpcache.Cache
+}
+
+// Topic represents the data which is cached for a topic to be used by the dp-search-data-extractor
+type Topic struct {
+	ID              string
+	LocaliseKeyName string
+	Slug            string
+	ReleaseDate     *time.Time
+	// Query is a comma separated string of topic id and its subtopic ids which will be used by the application to create the query
+	Query string
+	// List is a map[string]Subtopics which contains the topic id and a list of it's subtopics
+	List *Subtopics
+	// This is a reference to the parent topic
+	ParentID string
+}
+
+// NewTopicCache create a topic cache object to be used in the service which will update at every updateInterval
+// If updateInterval is nil, this means that the cache will only be updated once at the start of the service
+func NewTopicCache(ctx context.Context, updateInterval *time.Duration) (*TopicCache, error) {
+	config := dpcache.Config{
+		UpdateInterval: updateInterval,
+	}
+
+	cache, err := dpcache.NewCache(ctx, config)
+	if err != nil {
+		logData := log.Data{
+			"update_interval": updateInterval,
+		}
+		log.Error(ctx, "failed to create cache from dpcache", err, logData)
+		return nil, err
+	}
+
+	topicCache := &TopicCache{cache}
+
+	return topicCache, nil
+}
+
+func (dc *TopicCache) GetData(ctx context.Context, key string) (*Topic, error) {
```

Comment:
Missing godoc.
Also as elsewhere, the receiver var name mismatches. (Also for other methods in this file)
```suggestion
// GetData …blah… blah…
func (tc *TopicCache) GetData(ctx context.Context, key string) (*Topic, error) {
```